velox/exec/tests/CrossJoinTest.cpp

```diff
@@ -246,3 +246,33 @@ TEST_F(CrossJoinTest, parallelism) {
 
   OperatorTestBase::assertQuery(params, "VALUES (30), (30), (30), (30), (30)");
 }
+
+TEST_F(CrossJoinTest, parallelism2) {
+  // Setup 5 threads for build and probe. Each build thread gets 3 identical
+  // rows of input from the Values operator. The hashAgg operation results in
+  // a single row from these. The build thread that finishes last
+  // combines data from all other threads aggregating it to 1 row and puts them
+  // into the CrossJoinBridge. All probe threads get 2 identical rows of input
+  // from the Values operator and join them with 1 row of build side data from
+  // the bridge. Each probe thread is expected to produce 2 rows.
+
+  auto left = {makeRowVector({sequence<int32_t>(2)})};
+  auto right = {makeRowVector({sequence<int32_t>(3)})};
+
+  for (int i = 0; i < 10; i++) {
+    CursorParameters params;
+    params.maxDrivers = 5;
+    params.planNode = PlanBuilder(10)
+                          .values({left}, true)
+                          .crossJoin(
+                              PlanBuilder()
+                                  .values({right}, true)
+                                  .partialAggregation({}, {"min(1)"})
```

Comment:
BTW, it is not a valid plan to use partial aggregation as an input to a cross join.